### DISCUSSION TITLE
Should we add a section about the order of members to the styleguide?

### DISCUSSION THREAD

**Discussion by mgerhold** [mgerhold/silly-sprites#43] [General]
While @PurpurTentakel97 was reviewing [this code](https://github.com/mgerhold/silly-sprites/blob/d83f6d24fb68e2eca633052f3a2b3ad8e9b3d153/src/silly_sprites/entity.hpp#L9-L46), the question occurred whether we should add a section to the styleguide that defines the preferred ordering of class members and also whether they should be split between data members and member functions.
Let's use this thread to discuss the topic.

My personal suggestion:
* data members at the top (first public, then private)
* member functions below (first public, then private)
* the language forces us to put type aliases before we use them (order matters for those), so I'd suggest:
    * public type aliases before anything else
    * private type aliases wherever you want (but preferably at the top of the private section that needs them)

What are your opinions? :)

**Comment by PurpurTentakel97** [comment]
over all i agree with that.
things like in class enums need to be playced bevor two right?

I would suggest this
```
class Test final {

// forced
public:
protected:
private:

// data member
public:
protected:
private:

// member functions
public:
protected:
private:
}
```

there are opinions that put the public things wirst so you can see the interface.
but i like to have the sections as i mentioned.

Do we always write the label or only when we need them?
I would say only if we need them. otherways there would be a verry noisy code.
what do you think?

**Comment by PurpurTentakel97** [comment]
We should add the result of this discussion into the style guide.
